pkg/mount/raw_mount.go

```diff
@@ -0,0 +1,158 @@
+// +build linux
+
+package mount
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/docker/docker/pkg/mount"
+	"github.com/libopenstorage/openstorage/pkg/keylock"
+	"github.com/sirupsen/logrus"
+)
+
+// rawMounter loads mounts for raw volumes that are bind mounted in the mount table
+type rawMounter struct {
+	Mounter
+}
+
+// NewRawBindMounter returns a new rawBindMounter
+func NewRawBindMounter(
+	rootSubstrings []string,
+	mountImpl MountImpl,
+	allowedDirs []string,
+	trashLocation string,
+) (*rawMounter, error) {
+	b := &rawMounter{
+		Mounter: Mounter{
+			mountImpl:     mountImpl,
+			mounts:        make(DeviceMap),
+			paths:         make(PathMap),
+			allowedDirs:   allowedDirs,
+			kl:            keylock.New(),
+			trashLocation: trashLocation,
+		},
+	}
+	if err := b.Load(rootSubstrings); err != nil {
+		return nil, err
+	}
+	return b, nil
+}
+
+func (b *rawMounter) Reload(rootSubstring string) error {
+	newRBM, err := NewRawBindMounter(
+		[]string{rootSubstring},
+		b.mountImpl,
+		b.allowedDirs,
+		b.trashLocation,
+	)
+	if err != nil {
+		return err
+	}
+	return b.reload(rootSubstring, newRBM.mounts[rootSubstring])
+}
+
+// this mount filtering implementation is done based on logic implemented in findmnt + libmount
+func (b *rawMounter) Load(rawVolumeDevicesPaths []string) error {
+	mountPoints, err := GetMounts()
+	if err != nil {
+		return err
+	}
+
+	// try to find all bind mounts of raw volumes
+	if len(rawVolumeDevicesPaths) == 0 || rawVolumeDevicesPaths[0] == "" {
+		mountPointsByMajMin := make(map[string]*[]mount.Info)
+		mountPointsByTarget := make(map[string]*mount.Info)
+
+		for _, mp := range mountPoints {
+			// skip proc mount points
+			if strings.HasPrefix(mp.Mountpoint, "/proc") || mp.Root == "/null" {
```

Comment:
Maybe add these as a const string array of locations that need to be skipped so we can add more in the future if required.